### PR TITLE
[master][CC] Adding docs for deploying config changes into Choreo Connect

### PR DESCRIPTION
**Pull request by VajiraPrabuddhaka**
## Purpose

This PR will add a section under `Production Deployment Guide` of Choreo Connect to mention about how to properly deploy a config change into running instance of Choreo Connect.

### CODE REVIEW

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 1]
```diff
@@ -470,4 +470,16 @@ wso2.deployment.adapter.security.adapterRestService.enabled="false"
 
 #### Step 5: Deploy Choreo Connect
 
-Please follow the document about [Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts/) for deploying Choreo Connect.
\ No newline at end of file
+Please follow the document about [Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts/) for deploying Choreo Connect.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:473]:
```suggestion
Follow the document on [Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts/) for deploying Choreo Connect.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 2]
```diff
@@ -470,4 +470,16 @@ wso2.deployment.adapter.security.adapterRestService.enabled="false"
 
 #### Step 5: Deploy Choreo Connect
 
-Please follow the document about [Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts/) for deploying Choreo Connect.
\ No newline at end of file
+Follow the document on [Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts/) to deploy Choreo Connect.
+
+### Applying Config Changes into Running Instance of Choreo Connect
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:475]:
```suggestion
### Applying config changes into a running instance of Choreo Connect 
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 3]
```diff
@@ -470,4 +470,16 @@ wso2.deployment.adapter.security.adapterRestService.enabled="false"
...
-Please follow the document about [Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts/) for deploying Choreo Connect.
\ No newline at end of file
+Follow the document on [Deploying Choreo Connect as a Standalone Gateway on Kubernetes - Helm Artifacts]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-as-a-standalone-gateway-on-kubernetes-helm-artifacts/) to deploy Choreo Connect.
+
+### Applying config changes into a running instance of Choreo Connect 
+
+When you have to deploy a config change to the Choreo Connect running on production environment, we recommand you to complete the following steps in order.
+
+!!! Attention
+    You must follow this, if the change of the config is related to **enforcer** as `enforcer` is fetching configs from adapter only at the startup.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:480]:
```suggestion
    You must follow this, if the config change is related to the **Enforcer** as the `enforcer` fetches configs from the Adapter only at the startup.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 4]
```diff
@@ -470,4 +470,16 @@ wso2.deployment.adapter.security.adapterRestService.enabled="false"
...
+### Applying config changes into a running instance of Choreo Connect 
+
+When you have to deploy a config change to the Choreo Connect running on production environment, we recommand you to complete the following steps in order.
+
+!!! Attention
+    You must follow this, if the config change is related to the **Enforcer** as the `enforcer` fetches configs from the Adapter only at the startup.
+
+Steps:
+
+1. Do the config change and first rollout restart the adapter.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:484]:
```suggestion
1. Do the config change and first rollout restart the Adapter.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 5]
```diff
@@ -470,4 +470,16 @@ wso2.deployment.adapter.security.adapterRestService.enabled="false"
...
+
+When you have to deploy a config change to the Choreo Connect running on production environment, we recommand you to complete the following steps in order.
+
+!!! Attention
+    You must follow this, if the config change is related to the **Enforcer** as the `enforcer` fetches configs from the Adapter only at the startup.
+
+Steps:
+
+1. Do the config change and first rollout restart the Adapter.
+2. After all replicas of adapter pod are started and ready, rollout restart enforcer & router (`choreo-connect-deployment`).
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:485]:
```suggestion
2. After all the replicas of the Adapter pod have been started and are ready, rollout restart the Enforcer and Router (`choreo-connect-deployment`).
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md [hunk 6]
```diff
@@ -470,4 +470,16 @@ wso2.deployment.adapter.security.adapterRestService.enabled="false"
...
+
+When you have to deploy a config change to the Choreo Connect running on production environment, we recommand you to complete the following steps in order.
+
+!!! Attention
+    You must follow this, if the config change is related to the **Enforcer** as the `enforcer` fetches configs from the Adapter only at the startup.
+
+Steps:
+
+1. Do the config change and first rollout restart the Adapter.
+2. After all the replicas of the Adapter pod have been started and are ready, rollout restart the Enforcer and Router (`choreo-connect-deployment`).
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:485]: What do we mean by "rollout restart"? Are there some different steps to do this compared to normal restart?

**Review by samgnaniah — APPROVED** [review]

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md (hunk 6 above)
Comment (VajiraPrabuddhaka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:485]: So this production development guide is based on deploying things in kubernetes.. what I meant by "rolling restart" is the default deployment strategy of kubernetes.

Comment (samgnaniah) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:485]: Shall we add that context here?

Comment (pubudu538) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md:485]: Maybe we can link to https://kubernetes.io/docs/reference/kubectl/cheatsheet/#updating-resources

Comment (VajiraPrabuddhaka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md]: Linked to the above..

**Review by samgnaniah — APPROVED** [review]

**Pull request merged**